packages/manager/apps/pci-kubernetes/src/components/oidc/OidcProviderModal.component.tsx

```diff
@@ -140,118 +178,171 @@ export function OidcProviderModal() {
-              >
-                {t(
-                  'pci_projects_project_kubernetes_details_service_upsert_oidc_provider_field_url_extra',
-                )}
-              </OsdsText>
-              <Controller
-                name="issuerUrl"
-                control={control}
-                render={({ field: { onChange, onBlur, value } }) => (
-                  <OsdsInput
-                    type={ODS_INPUT_TYPE.text}
-                    className={`mt-3 ${errors.issuerUrl ? 'bg-red-100' : ''}`}
-                    data-testid="issuerUrl-input"
-                    onOdsInputBlur={onBlur}
-                    value={value}
-                    onOdsValueChange={onChange}
-                  />
-                )}
-              />
-            </OsdsFormField>
-            <OsdsFormField
-              class="mt-10"
-              data-testid="clientId-formfield"
-              error={t(errors.clientId?.message) || ''}
-            >
-              <OsdsText
-                size={ODS_TEXT_SIZE._200}
-                level={ODS_TEXT_LEVEL.body}
                 color={ODS_THEME_COLOR_INTENT.text}
-                slot="label"
               >
                 {t(
-                  `${mode}-oidc-provider:pci_projects_project_kubernetes_details_service_${mode}_oidc_provider_field_client_id`,
+                  'pci_projects_project_kubernetes_details_service_upsert_oidc_provider_description',
                 )}
               </OsdsText>
-              <OsdsText
-                size={ODS_TEXT_SIZE._400}
-                level={ODS_TEXT_LEVEL.body}
-                color={ODS_TEXT_COLOR_INTENT.text}
+
+              <OsdsFormField
+                class="mt-6"
+                data-testid="issuerUrl-formfield"
+                error={t(errors.issuerUrl?.message) || ''}
               >
-                {t(
-                  'pci_projects_project_kubernetes_details_service_upsert_oidc_provider_field_client_id_extra',
-                )}
-              </OsdsText>
-              <Controller
-                name="clientId"
-                control={control}
-                render={({ field: { onChange, onBlur, value } }) => (
-                  <OsdsInput
-                    type={ODS_INPUT_TYPE.text}
-                    className={`mt-3 ${errors.clientId ? 'bg-red-100' : ''}`}
-                    data-testid="clientId-input"
-                    onOdsInputBlur={onBlur}
-                    value={value}
-                    onOdsValueChange={onChange}
-                  />
-                )}
-              />
-            </OsdsFormField>
-            <div className="mt-6 flex justify-end gap-4">
-              <OsdsButton
-                color={ODS_THEME_COLOR_INTENT.primary}
-                variant={ODS_BUTTON_VARIANT.ghost}
-                onClick={onClose}
-                data-testid={`${mode}-oidc-provider:${mode}OIDCProvider-button_cancel`}
+                <OsdsText
```

Comment:
There are a lot of redundancies in these elements, please think to refactor code using high level functions and/or React.CloneELement to prevent from rewriting the same things with different props.